File: src/main/java/uk/co/squadlist/web/services/filters/ActiveMemberFilter.java
package uk.co.squadlist.web.services.filters;

import com.google.common.collect.Lists;
import org.springframework.stereotype.Component;
import uk.co.squadlist.model.swagger.Member;

import java.util.List;

@Component
public class ActiveMemberFilter {	// TODO suggests missing API functionality

	public List<Member> extractActive(List<Member> members) {
		List<Member> selected = Lists.newArrayList();
		for (Member member : members) {
			if (!member.isInactive()) {
				selected.add(member);
			}
		}
		return selected;
	}
	public List<Member> extractInactive(List<Member> members) {
		List<Member> selected = Lists.newArrayList();
		for (Member member : members) {
			if (member.isInactive()) {
				selected.add(member);
			}
		}
		return selected;
	}

}

File: src/main/kotlin/uk/co/squadlist/web/services/filters/ActiveMemberFilter.kt
package uk.co.squadlist.web.services.filters

import org.springframework.stereotype.Component
import uk.co.squadlist.model.swagger.Member

@Component
class ActiveMemberFilter {
    // TODO suggests missing API functionality
    fun extractActive(members: List<Member>): List<Member> {
        return members.filter { member -> !member.isInactive }
    }

    fun extractInactive(members: List<Member>): List<Member> {
        return members.filter { member -> member.isInactive }
    }

}